packages/test/test-end-to-end-tests/src/test/dataStoresNested.spec.ts

```diff
@@ -0,0 +1,194 @@
+		summaryConfigOverrides: {
+			state: "disableHeuristics",
+			maxAckWaitTime: 20000,
+			maxOpsSinceLastSummary: 7000,
+			initialSummarizerDelayMs: 0,
+		},
+	};
+
+	const runtimeOptions: IContainerRuntimeOptions = {
+		enableGroupedBatching: true,
+		summaryOptions: summaryOptionsToDisableHeuristics, // Force summarizer heuristics to be disabled so we can control when to summarize.
+	};
+
+	const testObjectFactory: TestFluidObjectFactory = new TestFluidObjectFactory(
+		[["test", new MapFactory()]],
+		"testObjectFactoryType",
+	);
+
+	const dataStoreFactory = new DataStoresFactory(
+		[[testObjectFactory.type, Promise.resolve(testObjectFactory)]],
+		async (runtime: IFluidDataStoreChannel) => {
+			return runtime;
+		},
+	);
+
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
+		defaultFactory: dataStoreFactory,
+		registryEntries: [[dataStoreFactory.type, Promise.resolve(dataStoreFactory)]],
+		runtimeOptions,
+	});
+
+	async function addContainer(container: IContainer) {
+		containers.push(container);
+		const dataStores = (await container.getEntryPoint()) as IDataStores;
+
+		await provider.ensureSynchronized();
+
+		return { container, dataStores };
+	}
+
+	const createContainer = async () => {
+		const container = await provider.createContainer(runtimeFactory);
+		return addContainer(container);
+	};
+
+	async function addContainerInstance() {
+		const container = await provider.loadContainer(runtimeFactory);
+		return addContainer(container);
+	}
+
+	beforeEach("getTestObjectProvider", async () => {
+		const driver = new LocalServerTestDriver();
+		const registry = [];
+		seed = 1; // Every test is independent from another test!
+
+		provider = new TestObjectProvider(
+			Loader,
+			driver,
+			() =>
+				new TestContainerRuntimeFactory(
+					"@fluid-experimental/test-dataStores",
+					new TestFluidObjectFactory(registry),
+				),
+		);
+		// syncSummarizer: true
+		provider.resetLoaderContainerTracker(true); // syncSummarizerClients
+
+		loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+	});
+
+	afterEach(() => {
+		provider.reset();
+		for (const container of containers) {
+			container.close();
+		}
+		containers = [];
+		summaryCollection = undefined;
+		summarizer = undefined;
+		loader = undefined;
+	});
+
+	async function waitForSummary(): Promise<string> {
+		// ensure that all changes made it through
+		await provider.ensureSynchronized();
+
+		assert(summaryCollection !== undefined, "summary setup properly");
+		// create promise before we call summarizeNow, as otherwise we might miss summary and will wait
+		// forever for next one to happen
+		const wait = summaryCollection.waitSummaryAck(
+			containers[0].deltaManager.lastSequenceNumber,
+		);
+		const summaryResult = await summarizeNow(summarizer!);
+		assert(summaryResult.summaryVersion !== undefined, "summary result");
+		const ackedSummary = await wait;
```

Comment:
Total side note, I think because of how summarizeNow already waits for the summary ack, there might be an advantage to just using await provider.ensureSynchronized. Feel free to just resolve this comment.

Reply:
I think if you do not need a new version from summary, then you are right. But some code wants to use version and load specific version. Not sure - it looks Ok as is :)